src/shared/generator.c

```diff
@@ -653,6 +653,12 @@ int generator_hook_up_growfs(
         if (empty_or_root(where)) {
                 growfs_unit = SPECIAL_GROWFS_ROOT_SERVICE;
                 growfs_unit_path = SYSTEM_DATA_UNIT_DIR "/" SPECIAL_GROWFS_ROOT_SERVICE;
+
+                /* If systemd-growfs-root.service is enabled, we also want it to be part of
+                 * root-fs-ready.target. */
+                r = generator_add_symlink(dir, SPECIAL_ROOT_FS_READY_TARGET, "wants", growfs_unit_path);
+                if (r < 0)
+                        return r;
```

Comment:
i am pretty sure that this should not be generated here, but simply be configured in systemd-growfs-root.service itself. after all botht root-fs-ready.target *and* systemd-growfs-root.service are just regular static services, hence they should not need us to generate the deps between them dynamically.